fulp_modules/QM_wrestling/code/wrestling.dm

```diff
@@ -0,0 +1,455 @@
+				if(NORTH)
+					target.pixel_x = user.pixel_x
+				if(SOUTH)
+					target.pixel_x = user.pixel_x
+				if(EAST)
+					target.pixel_x = user.pixel_x - 8
+				if(WEST)
+					target.pixel_x = user.pixel_x + 8
+			if(get_dist(user, target) > 1)
+				to_chat(user, "<span class='warning'>[target] is too far away!</span>")
+				user.pixel_x = user.base_pixel_x
+				user.pixel_y = user.base_pixel_y
+				target.pixel_x = target.base_pixel_x
+				target.pixel_y = target.base_pixel_y
+				return
+			if(!isturf(user.loc) || !isturf(target.loc))
+				to_chat(user, "<span class='warning'>You can't slam [target] here!</span>")
+				user.pixel_x = user.base_pixel_x
+				user.pixel_y = user.base_pixel_y
+				target.pixel_x = target.base_pixel_x
+				target.pixel_y = target.base_pixel_y
+				return
+		else
+			if(user)
+				user.pixel_x = user.base_pixel_x
+				user.pixel_y = user.base_pixel_y
+			if(target)
+				target.pixel_x = target.base_pixel_x
+				target.pixel_y = target.base_pixel_y
+			return
+		sleep(1)
+	if(user && target)
+		user.pixel_x = user.base_pixel_x
+		user.pixel_y = user.base_pixel_y
+		target.pixel_x = target.base_pixel_x
+		target.pixel_y = target.base_pixel_y
+
+		if(get_dist(user, target) > 1)
+			to_chat(user, "<span class='warning'>[target] is too far away!</span>")
+			return
+		if(!isturf(user.loc) || !isturf(target.loc))
+			to_chat(user, "<span class='warning'>You can't slam [target] here!</span>")
+			return
+		target.forceMove(user.loc)
+		var/fluff = "body-slam"
+		target.visible_message("<span class='danger'>[user] [fluff] [target]!</span>", \
+						"<span class='userdanger'>You're [fluff]ed by [user]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", COMBAT_MESSAGE_RANGE, user)
+		to_chat(user, "<span class='danger'>You [fluff] [target]!</span>")
+		playsound(user.loc, "swing_hit", 50, TRUE)
+		if(!target.stat)
+			target.emote("scream")
+			target.Paralyze(30)
+			target.adjustBruteLoss(rand(10,15))
+		else
+			target.adjustBruteLoss(rand(10,15))
+	else
+		if(user)
+			user.pixel_x = user.base_pixel_x
+			user.pixel_y = user.base_pixel_y
+		if(target)
+			target.pixel_x = target.base_pixel_x
+			target.pixel_y = target.base_pixel_y
+
+	log_combat(user, target, "body-slammed (Wrestling)")
+	return
+
+/datum/martial_art/fulpwrestling/proc/FlipAnimation(mob/living/D)
+	set waitfor = FALSE
+	if(D)
+		animate(D, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
+	if(do_after(D, 10 SECONDS))
+		if(D)
+			animate(D, transform = null, time = 1, loop = 0)
+
+/// Drop
+/datum/martial_art/fulpwrestling/proc/fulpdrop(mob/living/user, mob/living/target)
+	if(!can_use(user))
+		return FALSE
+	if(!target)
+		return
+	var/obj/surface = null
+	var/turf/ST = null
+	var/falling = 0
+
+	for(var/obj/O in oview(1, user))
+		if(O.density == 1)
+			if(O == user)
+				continue
+			if(O == target)
+				continue
+			if(O.opacity)
+				continue
+			else
+				surface = O
+				ST = get_turf(O)
+				break
```

Comment:
can this make you jump onto a table that is adjacent to you but not your target, causing the move to whiff even if your target stands perfectly still?

Reply:
I'd recommend removing the automatic climbing part of this move entirely and instead just giving wrestlers the passive ability to climb onto things much faster (check to see how the freerunning quirk does it, and maybe make it even faster than that) and not get stunned from doing so. then you can just make this move only executable if you're on a climbable structure or something and either remove the recoil part of it or make the recoil condition more lenient (maybe you can leap to reach a target that only ran X tiles away or something).